tasks/install_os.yml

```diff
@@ -25,10 +25,15 @@
     os_install: "RHEL 7.8"
   when: needed_os == "RHEL 7.9"
 
+- name: set os_install before updating os RHEL 8.4
+  set_fact:
+    os_install: "RHEL 8.2"
+  when: needed_os == "RHEL 8.4"
```

Comment:
Add a condition to check if the required os is already installed, if yes then it should not define os_install fact otherwise set the os_install to required RHEL version.